apple/t2/default.nix

```diff
@@ -11,45 +11,85 @@ let
-    replacements = [ "--replace" "/usr/bin/sed" "${pkgs.gnused}/bin/sed" ];
-  });
+  config = {
+    # For keyboard and touchbar
+    boot.kernelPackages = with pkgs; recurseIntoAttrs (linuxPackagesFor (callPackage ./pkgs/linux-t2.nix { }));
+    boot.initrd.kernelModules = [ "apple-bce" ];
 
-  hardware.pulseaudio.package = overrideAudioFiles pkgs.pulseaudio "src/modules/";
+    # For audio
+    boot.kernelParams = [ "pcie_ports=compat" "intel_iommu=on" "iommu=pt" ];
+    services.udev.extraRules = builtins.readFile (pkgs.substitute {
+      src = "${audioFiles}/files/91-audio-custom.rules";
+      replacements = [ "--replace" "/usr/bin/sed" "${pkgs.gnused}/bin/sed" ];
+    });
 
-  services.pipewire = rec {
-    package = overrideAudioFiles pkgs.pipewire "spa/plugins/";
+    hardware.pulseaudio.package = overrideAudioFiles pkgs.pulseaudio "src/modules/";
 
-    wireplumber.package = pkgs.wireplumber.override {
-      pipewire = package;
+    services.pipewire.package = pipewirePackage;
+    services.pipewire.wireplumber.package = pkgs.wireplumber.override {
+      pipewire = pipewirePackage;
+    };
+
+    # Make sure post-resume.service exists
+    powerManagement.enable = true;
+
+    systemd.services.fix-keyboard-backlight-and-touchbar = {
+      path = [ pkgs.kmod ];
+      serviceConfig.ExecStart = ''${pkgs.systemd}/bin/systemd-inhibit --what=sleep --why="fixing keyboard backlight and touchbar must finish before sleep" --mode=delay ${./fix-keyboard-backlight-and-touchbar.sh}'';
+      serviceConfig.Type = "oneshot";
+      description = "reload touchbar driver and restart upower";
+      wantedBy = [ "display-manager.service" "post-resume.target" ];
+      after = [ "post-resume.target" ];
     };
-  };
 
-  # Make sure post-resume.service exists
-  powerManagement.enable = true;
-
-  systemd.services.fix-keyboard-backlight-and-touchbar = {
-    path = [ pkgs.kmod ];
-    serviceConfig.ExecStart = ''${pkgs.systemd}/bin/systemd-inhibit --what=sleep --why="fixing keyboard backlight and touchbar must finish before sleep" --mode=delay ${./fix-keyboard-backlight-and-touchbar.sh}'';
-    serviceConfig.Type = "oneshot";
-    description = "reload touchbar driver and restart upower";
-    # must run at boot (and not too early), and after suspend
-    wantedBy = [ "display-manager.service" "post-resume.target" ];
-    # prevent running before suspend
-    after = [ "post-resume.target" ];
+    # Activation script to install apple-set-os-loader in order to unlock the iGPU 
+    system.activationScripts.appleSetOsLoader = lib.optionalString t2Cfg.enableAppleSetOsLoader ''
+      if [[ -e /boot/efi/EFI/BOOT/bootx64_original.efi ]]; then
+        true # It's already installed, no action required
+      elif [[ -e /boot/efi/EFI/BOOT/BOOTX64.EFI ]]; then
+        mv /boot/efi/EFI/BOOT/BOOTX64.EFI /boot/efi/EFI/BOOT/bootx64_original.efi
+        cp ${apple-set-os-loader-installer}/bootx64.efi /boot/efi/EFI/BOOT/bootx64.efi
```

Comment:
I would copy first the new bootloader to a temporary location, then mv the old bootloader to location with the original suffix and than the new bootloader to the final destination.
This is to reduce the disk that copying the new bootloader fails because the storage is full.

Reply:
Sorry for late reply! That's a great suggestion :-) 

I've implemented it now. Also made the iGPU activated by default.